src/utils/ucc_component.c

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2020.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+#include "config.h"
+#include "ucc_malloc.h"
+#include "ucc_component.h"
+#include "ucc_log.h"
+#include "core/ucc_global_opts.h"
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <err.h>
+#include <link.h>
+#include <dlfcn.h>
+#include <glob.h>
+#include <unistd.h>
+#include <string.h>
+
+static ucc_status_t ucc_component_load_one(const char *so_path,
+                                           const char *framework_name,
+                                           ucc_component_iface_t **c_iface)
+{
+    char                   iface_struct[128];
+    char                   pattern[128];
+    void                  *handle;
+    ucc_component_iface_t *iface;
+    int                    pos;
+
+    sprintf(pattern, "ucc_%s_", framework_name);
```

Comment:
snprintf or assert of strlen(framework_name)

Reply:
reworked this: added UCC_MAX_FRAMEWORK_NAME_LEN and added checks for strlen(framework_name).